src/main/java/pro/trevor/tankgame/state/board/Board.java

```diff
@@ -73,6 +73,16 @@ public Optional<IFloor> getFloor(Position p) {
         return getElementOnBoard(floorBoard, p);
     }
 
+    // Returns the unit at the position if there is one.
+    // If there is no unit at the position, then the floor is returned.
+    public Optional<IElement> getTopElement(Position p) {

```

Comment:
I see where the name getTopElement came from but it sounds like it's getting the Unit.  I think something like getUnitOrFloor is more descriptive.

Reply:
Why would it sound like that? Element != Unit. I did start with that name, but opted for this one instead. Not a big deal to switch back, but I want to understand why.

Reply:
Eh, I accept the other name actually, no need to respond.